oidc-service-impl/src/test/java/io/mosip/esignet/services/ConsentHelperServiceTest.java

```diff
@@ -374,6 +374,122 @@ public void processConsent_withEmptyConsent_thenPass(){
+        Claims normalizedClaims = new Claims();
+        normalizedClaims.setUserinfo(consentHelperService.normalizeClaims(claims.getUserinfo()));
+        normalizedClaims.setId_token(consentHelperService.normalizeClaims(claims.getId_token()));
+        String hashCode =consentHelperService.hashUserConsent(normalizedClaims,consentDetail.getAuthorizationScopes());
+        consentDetail.setHash(hashCode);
+
+        consentDetail.setAcceptedClaims(Arrays.asList("email","gender","name"));
+        consentDetail.setPermittedScopes(Arrays.asList("email","openid","profile"));
+
+        List<String> acceptedClaims = consentDetail.getAcceptedClaims();
+        List<String> permittedScopes = consentDetail.getPermittedScopes();
+        String jws = consentDetail.getSignature();
+        Collections.sort(acceptedClaims);
+        Collections.sort(permittedScopes);
+        Map<String,Object> payLoadMap = new TreeMap<>();
+        payLoadMap.put("accepted_claims",acceptedClaims);
+        payLoadMap.put("permitted_authorized_scopes",permittedScopes);
+
+        String signature = generateSignature(payLoadMap);;
+        consentDetail.setSignature(signature);
+        consentDetail.setPsuToken("psutoken");
+
+        PublicKeyRegistry publicKeyRegistry =new PublicKeyRegistry();
+        publicKeyRegistry.setCertificate(certificateString);
+        Mockito.when(authorizationHelperService.getIndividualId(oidcTransaction)).thenReturn("individualId");
+        Mockito.when(publicKeyRegistryService.findFirstByIdHashAndThumbprintAndExpiredtimes(Mockito.any(),Mockito.any())).thenReturn(Optional.empty());
+
+        Mockito.when(consentService.getUserConsent(userConsentRequest)).thenReturn(Optional.of(consentDetail));
+
+        consentHelperService.processConsent(oidcTransaction,true);
+
+        Assert.assertEquals(oidcTransaction.getConsentAction(),ConsentAction.CAPTURE);
+    }
+
+    @Test
+    public void processConsent_withInvalidSignature_thenFail(){
+
+        OIDCTransaction oidcTransaction=new OIDCTransaction();
+        oidcTransaction.setClientId("abc");
+        oidcTransaction.setPartnerSpecificUserToken("123");
+        oidcTransaction.setVoluntaryClaims(List.of("email"));
+        oidcTransaction.setEssentialClaims(List.of());
+        oidcTransaction.setRequestedAuthorizeScopes(List.of());
+        UserConsentRequest userConsentRequest = new UserConsentRequest();
+        userConsentRequest.setClientId(oidcTransaction.getClientId());
+        userConsentRequest.setPsuToken(oidcTransaction.getPartnerSpecificUserToken());
+
+        ConsentDetail consentDetail=new ConsentDetail();
+        consentDetail.setAcceptedClaims(Arrays.asList());
+        consentDetail.setSignature("haa.naa");
+
+        Mockito.when(consentService.getUserConsent(userConsentRequest)).thenReturn(Optional.of(consentDetail));
+
+        try{
+            consentHelperService.processConsent(oidcTransaction,true);
+            Assert.fail();
+        }catch (Exception e){}
```

Comment:
catch the proper exception and assert exception messages

Reply:
done